### DISCUSSION TITLE
Plugins: Publishe/Name, Display Name, Description

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6167] [Ideas]
![image](https://user-images.githubusercontent.com/830394/140299568-0a81eb9b-ae1d-4dc9-8649-176888fb45b4.png)

Plugins have three descriptive parameters:

1. `$:/plugins/publisher/name`  e.g. publisher/name
2. name field: acts as display name
3. description: some explanation on what plugin does

Idea: 
* The plugin information shows `Name: Description` which is useful
* But, in the sidebar only description is shown which is confusing

Proposal
Please use the same format of `Name: Description` in the Sidebar > More > Plugins e.g
![image](https://user-images.githubusercontent.com/830394/140300273-60adc3c3-f2e9-4c4b-a5b0-f3cf8bf53236.png)

I mean in the `Sidebar > More > Plugins` use name in bold and description in regular font

**Comment by nilslindemann** [comment]
@kookma 

To get the desired format, I overwrote [$:/core/ui/PluginListItemTemplate](https://tiddlywiki.com/#%24%3A%2Fcore%2Fui%2FPluginListItemTemplate) with this content:

```
<div class="tc-menu-list-item">
<$link to={{!!title}}>
''<$text text={{{ [<currentTiddler>get[name]] ~[<currentTiddler>split[/]last[1]] }}}/>'':
<$view field="description"><$view field="title"/></$view>
</$link>
</div>
```

The third line is copied and pasted from [$:/core/ui/Components/plugin-info](https://github.com/Jermolene/TiddlyWiki5/blob/122de63b71a2454314365ac396e051ad1db77641/core/ui/Components/plugin-info.tid#L42).

Maybe Jermolene did not do it because the links can get too long, or because the calculation of the bold text may last long when many plugins are installed. Or maybe he just forgot to add it. I also think it looks better. It could be refined to not contain the text after the bold text if it is the same. But to be honest, I currently don't know how to do that.

**Reply by nilslindemann** [reply]
I just see, this template is also used for the "Languages" and "Themes" tabs. Languages can e.g. be tested in [pmarios German translation](https://tiddlywiki.com/languages/de-DE/index.html#%24%3A%2Fcore%2Fui%2FPluginListItemTemplate)

**Comment by kookma** [comment]
Thanks @nilslindemann 
I hope @Jermolene could have a look and lets us his opinion!

I am tested your solution and it looks fine to me
![image](https://user-images.githubusercontent.com/830394/142732960-fe75a584-0634-40f3-ad7a-9eb821b5aa06.png)

I may also suggest the sort by title/name

**Reply by nilslindemann** [reply]
> I may also suggest the sort by title/name

Good idea, @kookma.

To achieve this, replace the text `sort[description]` with `sort[]` [in the first three tiddlers](https://tiddlywiki.com/#%24%3A%2Fcore%2Fui%2FMoreSideBar%2FPlugins%2FLanguages:%24%3A%2Fcore%2Fui%2FMoreSideBar%2FPlugins%2FLanguages%20%24%3A%2Fcore%2Fui%2FMoreSideBar%2FPlugins%2FPlugins%20%24%3A%2Fcore%2Fui%2FMoreSideBar%2FPlugins%2FTheme%20%24%3A%2Fcore%2Fui%2FPluginListItemTemplate), and apply above tweak to the last.

Here, `sort[]` is the shortcut for `sort[title]`, see the [sort Operator docs](https://tiddlywiki.com/#sort%20Operator).

**Reply by nilslindemann** [reply]
Gosh, I love the permaview button :-)

**Reply by kookma** [reply]
I got a little different work: For $:/core/ui/MoreSideBar/Plugins/Plugins for example:

```
\define xtitle() get[title]split[/]last[1]
<$list filter="[!has[draft.of]plugin-type[plugin]] :sort:string:caseinsensitive[get[name]else<xtitle>]" template="$:/core/ui/PluginListItemTemplate" emptyMessage={{$:/language/ControlPanel/Plugins/Empty/Hint}}>>/>
```

**Reply by nilslindemann** [reply]
@kookma Yes, even better, to sort case-insensitive.

BTW, the `>>` before the `/>` is a typo and can be removed, @Jermolene

**Reply by saqimtiaz** [reply]
I support this change, the listing in the side bar is pretty confusing without plugin names. I have been working around it by redundantly adding the plugin name to the beginning of the description as well.

**Comment by Jermolene** [comment]
Thanks @kookma I think we should go a bit further and use a miniaturised version of the plugin badge that we use in the control panel, just with the icon, name and description, on a white background in a black bordered rectangle. We should be able to reuse some of the existing styles.

**Comment by Jermolene** [comment]
I made a quick implementation of the [idea above](https://github.com/Jermolene/TiddlyWiki5/discussions/6167#discussioncomment-1707325):

<img width="432" alt="image" src="https://user-images.githubusercontent.com/174761/145035846-7974db92-d98e-4758-95dc-0f45459853a1.png">

The scope of the changes is so narrow that I think it's safe to merge for v5.2.1

**Reply by saqimtiaz** [reply]
That is a significant usability improvement!

**Reply by pmario** [reply]
hmmm, There is no version info anymore. .. this isn't a problem for the core plugins, but it is one for 3rd party plugins

**Reply by Jermolene** [reply]
@pmario there has never been version info shown in the sidebar. I did try including the title and version here but it just looks too crowded.

**Reply by pmario** [reply]
uuups, It's the sidebar view. Sorry ignore my comment

**Reply by nilslindemann** [reply]
Looks awesome @Jermolene !

**Reply by kookma** [reply]
Thank you @Jermolene, nice improvement!

There few issues however

- In Centralized theme, the plugin boxes overflow from the right without any scrollbar
- I feel the shadowed boxes make the sidebar a bit crowded! So, it may be more readable when Vanilla theme is used instead of Snow white
- The contrast is very low in some official  palettes like "Dark Photos" and "Muted"
<img src="https://user-images.githubusercontent.com/830394/145092335-49952998-9616-493e-8c07-9345fe68b045.png" width=320px/>

**Reply by Jermolene** [reply]
Thanks @kookma 

> * In Centralized theme, the plugin boxes overflow from the right without any scrollbar

Ouch! That actually seems like it might be a wider problem – the sidebar "tools" tab suffers from the same problem. Would you kindly open another issue for this?

> * I feel the shadowed boxes make the sidebar a bit crowded! So, it may be more readable when Vanilla theme is used instead of Snow white

Hmmm I like that the plugin info panels are consistent across the UI. Perhaps it would look better with more vertical space in between?

> * The contrast is very low in some official  palettes like "Dark Photos" and "Muted"

Thanks, fix upcoming.

**Reply by kookma** [reply]
Hi Jeremy
I will create another issue for centerlized theme.
I am fine with other items raised here.
Thank you
